Runtime/Behaviours/CinemachineStoryboard.cs

```diff
@@ -152,33 +210,18 @@ void CameraUpdatedCallback(CinemachineBrain brain)
             if (ci != null && ci.mCanvas != null)
                 ci.mCanvas.SetActive(showIt);
         }
-
+        
         CanvasInfo LocateMyCanvas(CinemachineBrain parent, bool createIfNotFound)
         {
             CanvasInfo ci = null;
             for (int i = 0; ci == null && i < mCanvasInfo.Count; ++i)
                 if (mCanvasInfo[i].mCanvasParent == parent)
                     ci = mCanvasInfo[i];
-            if (createIfNotFound)
+            if (createIfNotFound && ci == null)
             {
-                if (ci == null)
-                {
-                    ci = new CanvasInfo() { mCanvasParent = parent };
-                    int numChildren = parent.transform.childCount;
-                    for (int i = 0; ci.mCanvas == null && i < numChildren; ++i)
-                    {
-                        RectTransform child = parent.transform.GetChild(i) as RectTransform;
-                        if (child != null && child.name == CanvasName)
-                        {
-                            ci.mCanvas = child.gameObject;
-                            ci.mViewport = ci.mCanvas.GetComponentsInChildren<RectTransform>()[1]; // 0 is mCanvas
-                            ci.mRawImage = ci.mCanvas.GetComponentInChildren<UnityEngine.UI.RawImage>();
-                        }
-                    }
-                    mCanvasInfo.Add(ci);
-                }
```

Comment:
What was the reason for checking this? Is it for a use case with timeline?

Reply:
I don't understand the question.  Checking what?

Reply:
I deleted this part (165-179), because I don't see the point of it. 

If (CanvasInfo) ci is null, then it checks if brain has RectTrasform children, if it does, then it assings one to ci. 

Instead I do, if ci is null, create a canvas, because if there was a canvas before it was deleted by DestroyCanvas.
When could I be wrong with the assumption "if there was a canvas before it was deleted by DestroyCanvas".

Reply:
I did not find a case in my scenes but I have not checked with timeline. I thought maybe in timeline there is a use case. I don't really know timeline.

Reply:
Thank you 👍